tests/tests_integration/test_api/test_three_d.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import (
+    ThreeDAssetMapping,
+    ThreeDModelRevision,
+    ThreeDModelRevisionUpdate,
+    ThreeDModelUpdate,
+)
+
+COGNITE_CLIENT = CogniteClient()
+
+
+@pytest.fixture(scope="class")
+def test_revision():
+    model_id = COGNITE_CLIENT.three_d.models.list(limit=1)[0].id
+    revision = COGNITE_CLIENT.three_d.revisions.list(model_id=model_id, limit=1)[0]
+    return revision, model_id
+
+
+@pytest.fixture(scope="class")
+def new_model():
+    res = COGNITE_CLIENT.three_d.models.create(name="NewTestModel")
+    yield res
+    COGNITE_CLIENT.three_d.models.delete(id=res.id)
+    assert COGNITE_CLIENT.three_d.models.retrieve(id=res.id) is None
+
+
+@pytest.fixture(scope="class")
+def new_asset_mapping(test_revision):
+    revision, model_id = test_revision
+
+    node_id = COGNITE_CLIENT.three_d.revisions.list_nodes(model_id=model_id, revision_id=revision.id)[0].id
+    asset_id = COGNITE_CLIENT.assets.list(limit=1)[0].id
+    asset_mapping = ThreeDAssetMapping(node_id=node_id, asset_id=asset_id)
+    res = COGNITE_CLIENT.three_d.asset_mappings.create(
+        model_id=model_id, revision_id=revision.id, asset_mapping=asset_mapping
+    )
```

Comment:
You have a race condition here. Since we are running these integration tests in parallell for three different python versions, you will have this mapping be created and deleted 3 times. One of two things will happen:

1) API complains that mapping already exists
2) Test using this fixture will fail since another integration test will have deleted the mapping

Either drop this integration test or find a more robust of handling it. Nothing of the latter comes to mind immediately.